miio/waterheater.py

```diff
@@ -0,0 +1,219 @@
+            "Device cleaning is required: {result.cleaning_required}\n"
+            "Operation mode type: {result.mode.name} ({result.mode.value}) \n"
+            "Operation start time in booking mode: {result.booking_time_start}\n"
+            "Operation end time in booking mode: {result.booking_time_end}\n",
+        )
+    )
+    def status(self) -> WaterHeaterStatus:
+        """Retrieve properties."""
+        properties = SUPPORTED_MODELS.get(
+            self.model, SUPPORTED_MODELS[MODEL_WATERHEATER_E1]
+        )["available_properties"]
+
+        # A single request is limited to 16 properties. Therefore the
+        # properties are divided into multiple requests
+        _props_per_request = 15
+
+        # The E1 is limited to a single property per request
+        if self.model in [MODEL_WATERHEATER_E1]:
+            _props_per_request = 1
+
+        values = self.get_properties(properties, max_properties=_props_per_request)
+
+        return WaterHeaterStatus(dict(zip(properties, values)))
+
+    @command(default_output=format_output("Powering on"))
+    def on(self):
+        """Power on."""
+        return self.send("set_power", [1])
+
+    @command(default_output=format_output("Powering off"))
+    def off(self):
+        """Power off."""
+        return self.send("set_power", [0])
+
+    @command(
+        click.argument("temperature", type=int),
+        default_output=format_output("Setting target temperature to {temperature}"),
+    )
+    def set_target_temperature(self, temperature: int):
+        """Set target temperature."""
+        min_temp: int
+        max_temp: int
+        min_temp, max_temp = SUPPORTED_MODELS.get(
+            self.model, SUPPORTED_MODELS[MODEL_WATERHEATER_E1]
+        )["temperature_range"]
+        if not min_temp <= temperature <= max_temp:
+            raise WaterHeaterException("Invalid target temperature: %s" % temperature)
+        return self.send("set_temp", [temperature])
+
+    @command(default_output=format_output("Setting bacteriostatic mode on"))
+    def set_bacteriostatic_mode(self):
+        """Set bacteriostatic operational mode."""
+        mode: OperationMode
+        bacteriostatic_mode: bool
+        bacteriostatic_temp: int
+        bacteriostatic_mode = SUPPORTED_MODELS.get(
+            self.model, SUPPORTED_MODELS[MODEL_WATERHEATER_E1]
+        )["bacteriostatic_mode"]
+        if not bacteriostatic_mode:
+            raise WaterHeaterException("Bacteriostatic mode is not supported.")
+        bacteriostatic_temp = SUPPORTED_MODELS.get(
+            self.model, SUPPORTED_MODELS[MODEL_WATERHEATER_E1]
+        )["bacteriostatic_temperature"]
+        mode = OperationMode(self.send("get_prop", ["modeType"])[0])
+        if mode == OperationMode.Termostatic:
+            raise WaterHeaterException("Bacteriostatic operational mode is not supported in Termostatic mode.")
+        return self.send("set_temp", [bacteriostatic_temp])
```

Comment:
```suggestion
        )["bacteriostatic_mode"]
        
        if not bacteriostatic_mode:
            raise WaterHeaterException("Bacteriostatic mode is not supported.")
            
        bacteriostatic_temp = SUPPORTED_MODELS.get(
            self.model, SUPPORTED_MODELS[MODEL_WATERHEATER_E1]
        )["bacteriostatic_temperature"]
        
        
        mode = OperationMode(self.send("get_prop", ["modeType"])[0])
        if mode == OperationMode.Termostatic:
            raise WaterHeaterException("Bacteriostatic operational mode is not supported in Termostatic mode.")
        return self.send("set_temp", [bacteriostatic_temp])
```
Please add some newlines to separate logical code blocks for easier readability, also on `set_appoint` and `set_mode` :-)